src/hooks/eip4626/useReadTokenDecimals.ts

```diff
@@ -1,12 +1,14 @@
 import { useQuery } from "@tanstack/react-query";
 import { readTokenDecimals } from "@/services/contracts/wagmiGenActions";
 import { EvmAddress } from "@/types/types";
+import { QueryKeys } from "@/hooks/types";
 
 export function useReadTokenDecimals(tokenAddress?: EvmAddress) {
   return useQuery({
-    queryKey: [],
+    queryKey: [QueryKeys.ReadTokenDecimals, tokenAddress],
     queryFn: () => readTokenDecimals({}, tokenAddress),
     enabled: !!tokenAddress,
     select: (data) => Number(data),
+    initialData: () => 18,
```

Comment:
The same comment as i submitted above for `initialData()`.

Reply:
Same motivation, instead of expecting `undefined` in our formatter, we just use default until the actual decimals are fetched.
We can discard initial data in the future when having some component that deals with states and prevent potential cases when decimals are not arrived yet but we already try to format the amounts.